07_hello_friend/friend.rb

```diff
@@ -0,0 +1,6 @@
+class Friend
+  def greeting(str = '')
+    greet = 'Hello'
+    str == '' ? "#{greet}!" : "#{greet}, #{str}!"
```

Comment:
Here's a trick https://til.hashrocket.com/posts/f4agttd8si-chaining-then-in-ruby-26.

Also, you can use [str.empty?](https://ruby-doc.org/core-3.0.1/String.html#method-i-empty-3F).

Reply:
Refactored `str == ''` into `str.empty?` so that it matches Ruby best practices but besides using ternary operator can't convert if - then - else block into some other construct.

Reply:
Try this:
```ruby
'Hello!'.then { |greet| str.empty? ? greet : "#{greet}, #{str}!" }
```